ui/src/main/webapp/src/app/components/reports/application-index/application-index.component.ts

```diff
@@ -0,0 +1,223 @@
+    categoriesMultiStats: any[] = [];
+    mandatoryMultiStats: any[] = [];
+
+    constructor(
+        private _router: Router,
+        private _activatedRoute: ActivatedRoute,
+        private _notificationService: NotificationService,
+        private _aggregatedStatsService: AggregatedStatisticsService,
+        private _http: Http
+    ) { }
+
+
+    ngOnInit(): void {
+        this._activatedRoute.params.subscribe(params => {
+            let executionId = parseInt(params['executionId']);
+            this.execID = executionId;
+
+
+            this._aggregatedStatsService.getAggregatedCategories(executionId).subscribe(
+                result => this.categoriesMultiStats = this.getCategoriesStats(this.calculateCategoryIncidents(result), this.calculateStoryPointsInCategories(result)),
+                error => {
+                    this._notificationService.error(utils.getErrorMessage(error))
+                    this._router.navigate(['']);
+                }
+            );
+            this._aggregatedStatsService.getMandatoryIncidents(executionId).subscribe(
+                result => this.mandatoryMultiStats = this.getCategoryDetailsStats(this.convertToChartStatistic(result)),
+                error => {
+                    this._notificationService.error(utils.getErrorMessage(error))
+                    this._router.navigate(['']);
+                }
+            );
+            this._aggregatedStatsService.getAggregatedJavaPackages(executionId).subscribe(
+                result => this.globalPackageUseData = this.convertPackagesToChartStatistic(result),
+                error => {
+                    this._notificationService.error(utils.getErrorMessage(error))
+                    this._router.navigate(['']);
+                }
+            );
+        });
+
+        this._activatedRoute.parent.parent.parent.data.subscribe((data: { applicationGroup: ApplicationGroup }) => {
+            this.group = data.applicationGroup;
+        });
+    }
+
+    /**
+     * Array of package statistics converted into ChartStatistic array for table and chart rendering
+     */
+    private convertPackagesToChartStatistic(statistics): ChartStatistic[] {
+        if (!statistics)
+            return;
+
+        let result: ChartStatistic[] = [];
+
+        Object.keys(statistics).forEach(key => {
+            let chartStat: ChartStatistic = new ChartStatistic(key, statistics[key]);
+            result.push(chartStat);
+        });
+        result = result.sort(chartStatisticComparator);
+        return result;
+    }
+
+    /**
+     * Converting array of categories statistics into ChartStatistic array for table and chart rendering
+     */
+    private convertToChartStatistic(statistics): ChartStatistic[] {
+        if (!statistics)
+            return;
+
+        let result: ChartStatistic[] = [];
+
+        Object.keys(statistics).forEach(key => {
+            var incidents = statistics[key];
```

Comment:
actually probably from some old memories. It confuses me let vs var declaration too, there is no specific reason. I read the differences between these too and will sync them to let at the end.